ParProcCo/job_scheduler.py

```diff
@@ -265,56 +254,55 @@ def _submit_jobs(
         try:
             for job_scheduling_info in job_scheduling_info_list:
                 logging.debug(
-                    "Submitting job on cluster for"
-                    f" job script {job_scheduling_info.job_script_path}"
-                    f" and args {job_scheduling_info.job_script_arguments}"
+                    "Submitting job on cluster for job script '%s' and args '%s'",
+                    job_scheduling_info.job_script_path,
+                    job_scheduling_info.job_script_arguments,
                 )
                 submission = self.make_job_submission(job_scheduling_info)
                 assert submission.job is not None
-                resp = self.client.submit_job(submission)
-                if resp.job_id is None:
+                resp = None
+                try:
                     resp = self.client.submit_job(submission)
-                    if resp.job_id is None:
-                        raise ValueError("Job submission failed", resp.errors)
+                except Exception:
+                    logging.error("Job submission failed. Trying again", exc_info=True)
+
+                if resp is None or resp.job_id is None:
+                    try:
+                        resp = self.client.submit_job(submission)
+                    except Exception:
+                        logging.error("Job submission failed", exc_info=True)
+                        raise ValueError("Job submission failed")
+                assert resp.job_id
                 job_scheduling_info.set_job_id(resp.job_id)
                 job_scheduling_info.update_status_info(
                     StatusInfo(
                         submit_time=datetime.now(),
                     )
                 )
-                logging.debug(
-                    f"Job for job script {job_scheduling_info.job_script_path}"
-                    f" and args {submission.job.argv} has been submitted with"
-                    f" id {resp.job_id}"
+                logging.info(
+                    "Submitted job with id %d",
+                    resp.job_id,
                 )
```

Comment:
This is the only change I'm not sure about - I'd prefer to have more than the job id reported, as I find it useful to have a link between the job id and what it's doing somewhere in the logs. That said, perhaps `job_scheduling_info.job_name` would be more reasonable than logging the full command?
```suggestion
                logging.info(
                    "Submitted job '%s' with id %d",
                    job_scheduling_info.job_name,
                    resp.job_id,
                )
```